odl/solvers/iterative/iterative.py

```diff
@@ -290,6 +289,106 @@ def conjugate_gradient_normal(op, x, rhs, niter=1, callback=None):
             callback(x)
 
 
+def conjugate_gradient_nonlinear(f, x, rhs, niter=1, nreset=0,
+                                 line_search=None, tol=1e-16, beta_method='FR',
+                                 callback=None):
+    """Conjugate gradient for nonlinear problems.
+
+    The method is described in a
+    `Wikipedia article
+    <https://en.wikipedia.org/wiki/Nonlinear_conjugate_gradient_method>`_.
```

Comment:
If you want a book reference there is one in GNS2009 (at least for what wiki calls Flethcer-Reeves update) which we already have in our list of references